internal/models/reservation_model.go

```diff
@@ -80,6 +80,41 @@ type AWSReservation struct {
 	Detail *AWSDetail `db:"detail" json:"detail"`
 }
 
+type GCPDetail struct {
+	Zone string `json:"zone"`
+
+	// Optional instance name
+	Name *string `json:"name"`
+
+	// GCP Machine type.
+	MachineType string `json:"machine_type"`
+
+	// Amount of instances to provision of type: Instance type.
+	Amount int64 `json:"amount"`
+
+	// Immediately power off the system after initialization
+	PowerOff bool `json:"poweroff"`
+}
+
+type GCPReservation struct {
+	Reservation
+
+	// Pubkey ID.
+	PubkeyID string `db:"pubkey_id" json:"pubkey_id"`
+
+	// Source ID.
+	SourceID string `db:"source_id" json:"source_id"`
+
+	// The ID of the aws reservation which was created.
+	GCPReservationID string `db:"gcp_reservation_id" json:"gcp_reservation_id"`
```

Comment:
I believe this is unused, I suggest to drop or comment out unused fields for now so that we won't forget to remove them in the future if they end up not being used at all.

Reply:
This is something I am still trying to figure out. I don't get the instance and reservation info right away

Reply:
It's okay to leave it here just put "TODO" here. Or maybe do not export this via OpenAPI so QA folks are not confused why it's blank.